src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs

```diff
@@ -4,16 +4,21 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
+using System.IO;
 using System.Text.RegularExpressions;
 
 using Microsoft.CodeAnalysis.Sarif.Writers;
 
+using Newtonsoft.Json;
+
 namespace Microsoft.CodeAnalysis.Sarif.Sarifer
 {
     [Export(typeof(IBackgroundAnalyzer))]
     internal class SpamBackgroundAnalyzer : BackgroundAnalyzerBase
     {
-        private readonly List<SpamRule> rules;
+        private readonly List<SpamRule> Rules = new List<SpamRule>();
+
+        private string CurrentSolutionDirectory = string.Empty;
```

Comment:
>string.Empty [](start = 50, length = 12)

Use `null`, because that's what the base class supplies as the `solutionDirectory` parameter to `AnalyzeCore` if there is no open solution.